multicluster/controllers/multicluster/commonarea/labelidentityimport_controller.go

```diff
@@ -0,0 +1,165 @@
+		namespace:             namespace,
+		remoteCommonArea:      remoteCommonArea,
+		installedLabelImports: cache.NewIndexer(resourceImportIndexerKeyFunc, cache.Indexers{}),
+	}
+}
+
+func resourceImportIndexerKeyFunc(obj interface{}) (string, error) {
+	ri := obj.(multiclusterv1alpha1.ResourceImport)
+	return common.NamespacedName(ri.Namespace, ri.Name), nil
+}
+
+func (r *LabelIdentityResourceImportReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	klog.V(2).InfoS("Reconciling LabelIdentity kind of ResourceImport", "resourceImport", req.NamespacedName)
+	if r.localClusterClient == nil {
+		return ctrl.Result{}, errors.New("localClusterClient has not been initialized properly, no local cluster client")
+	}
+	if r.remoteCommonArea == nil {
+		return ctrl.Result{}, errors.New("remoteCommonArea has not been initialized properly, no remote common area")
+	}
+	var labelIdentityResImport multiclusterv1alpha1.ResourceImport
+	err := r.remoteCommonArea.Get(ctx, req.NamespacedName, &labelIdentityResImport)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			klog.ErrorS(err, "Unable to fetch LabelIdentity kind of ResourceImport", "resourceImport", req.NamespacedName)
+			return ctrl.Result{}, err
+		}
+		labelIdentityResImportObj, exist, _ := r.installedLabelImports.GetByKey(req.NamespacedName.String())
+		if !exist {
+			// Stale controller will clean up any stale LabelIdentities
+			klog.ErrorS(err, "No cached data for LabelIdentity kind of ResourceImport", "resoureImport", req.NamespacedName)
+			return ctrl.Result{}, nil
+		}
+		labelIdentityResImport = labelIdentityResImportObj.(multiclusterv1alpha1.ResourceImport)
+		return r.handleLabelIdentityImpDelete(ctx, &labelIdentityResImport)
+	}
+	return r.handleLabelIdentityImpCreateOrUpdate(ctx, &labelIdentityResImport)
+}
+
+func (r *LabelIdentityResourceImportReconciler) handleLabelIdentityImpCreateOrUpdate(ctx context.Context,
+	labelResImp *multiclusterv1alpha1.ResourceImport) (ctrl.Result, error) {
+	labelIdentityName := types.NamespacedName{
+		Name: labelResImp.Name,
+	}
+	labelIdentity := &multiclusterv1alpha1.LabelIdentity{}
+	err := r.localClusterClient.Get(ctx, labelIdentityName, labelIdentity)
+	labelIdentityNotFound := apierrors.IsNotFound(err)
+	if err != nil && !labelIdentityNotFound {
+		return ctrl.Result{}, err
+	}
+	if labelIdentityNotFound {
+		newLabelIdentity := &multiclusterv1alpha1.LabelIdentity{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: labelResImp.Name,
+			},
+			Spec: multiclusterv1alpha1.LabelIdentitySpec{
+				Label: labelResImp.Spec.LabelIdentity.Label,
+				ID:    labelResImp.Spec.LabelIdentity.ID,
+			},
+		}
+		if err = r.localClusterClient.Create(ctx, newLabelIdentity, &client.CreateOptions{}); err != nil {
+			klog.ErrorS(err, "Failed to create LabelIdentity", "clusterID", r.localClusterID, "label", labelResImp.Spec.LabelIdentity.Label)
+			return ctrl.Result{}, err
+		}
+		r.installedLabelImports.Add(*labelResImp)
```

Comment:
Do we really need `installedLabelImports`? I saw it is checked only before deleting an import. I feel that check is not very useful (in most case ResourceImport deletion should delete the IdentityImport, so the check does not save much).

Reply:
Again trying to be consistent with resourceimport_controller

Reply:
I do not think all controllers need to be the same. If installedLabelImports is not helpful for this one (I feel not, and removing it maybe even safer to remove the stale cache case in deletion), we should remove it.

Reply:
Done. Removed `installedLabelImports` for this file and also fixed a bug in `resourceimport_controller` where deleted resImps are never removed from cache @luolanzone